### PR TITLE
Feat/#49 generate smart contract with log data only

### LINKED ISSUE #13: as a smart contract developer, I want to setup a mechanism to store hash value only.
This feature comes from the idea that we can just only store document hash or even ID only; the full-document we stored in the transaction log. We can apply this feature on EOS blockchain for sure, but I am not quite sure about ICON or another blockchain. The benefit of this feature is that we can gain blockchain advantages but save the cost of saving data on the blockchain network.

### LINKED ISSUE #49: As a User, I want to create smart contract to log data only, without store anything to state data

### PR DESCRIPTION
**Pull request by manh-vv**
close #13

### CODE REVIEW

**Review by manh-vv — COMMENTED** [review]
LGTM

**Review by hoang-dao — COMMENTED** [review]
As discussed, we should update the ICON sm to check log data condition, @quocle108 can you help updating? Thanks.

There's a problem with `Logger` type in `packages/demux-icon/test/index.spec.ts` that makes our tests failed:
```
Type 'Logger' is missing the following properties from type 'Logger': addStream, addSerializers, child, reopenFileStreams, and 21 more.

    const iconBlock = new IconBlock(rawBlock, logger);
```
since we fixed logger dependency, we should fix that too.

The rest LGTM.

File: packages/abstraction/package.json [hunk 1]
```diff
@@ -27,7 +27,6 @@
   },
   "dependencies": {
     "@aloxide/bridge": "^0.1.0",
-    "@aloxide/logger": "^0.1.0",
```
[the remark this answers is not in the document]
Comment (manh-vv) [reply packages/abstraction/package.json:30]: Thank @hoang-dao

File: packages/example-generate-sm/src/index.ts [hunk 2]
```diff
@@ -1,15 +1,21 @@
 import { ContractGenerator } from '@aloxide/abstraction';
-import { EOSContractAdapter } from '@aloxide/bridge';
+import { EOSContractAdapter, ICONContractAdapter, ModelContractAdapter } from '@aloxide/bridge';
 import { createLogger } from 'bunyan';
 import path from 'path';
 
+const resultPath = path.resolve(__dirname, '../out');
```
Comment (tiendq) [inline packages/example-generate-sm/src/index.ts:6]: `outputPath` is more appropriate name, as it's also consistent with line 15.

File: packages/bridge/src/AbsContractAdapter.ts [hunk 3]
```diff
@@ -16,19 +17,32 @@ export abstract class AbsContractAdapter implements ContractAdapter {
...
+    this.config = config || {};
+    this.blockchainType = this.config.blockchainType;
+    this.logDataOnly = this.config.logDataOnly;
+  }
 
   generate(outputPath: string) {
-    this.logger.debug(`output path is: ${outputPath}, blockchain type: ${this.blockchainType}`);
-
     // read template from node_modules folder
     this.templatePath = path.resolve(
```
Comment (tiendq) [inline packages/bridge/src/AbsContractAdapter.ts:32]: Could it be initialized once in constructor?

File: packages/bridge/src/ContractAdapter.ts [hunk 4]
```diff
@@ -1,9 +1,18 @@
...
   entityConfigs: EntityConfig[];
   logger?: Logger;
+
+  /**
+   * With this option enabled, the generated smart-contract will not store any data to state-data
+   * https://github.com/lecle/aloxide/issues/49
+   *
+   * Default is false
+   */
+  logDataOnly?: boolean;
```
Comment (tiendq) [inline packages/bridge/src/ContractAdapter.ts:15]: Boolean variable isn't necessary optional if we have default value for it e.g. `false` here.

File: packages/abstraction/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (manh-vv) [reply packages/abstraction/package.json:30]: fixed

File: packages/example-generate-sm/src/index.ts (hunk 2 above)
Comment (manh-vv) [reply packages/example-generate-sm/src/index.ts]: fixed

File: packages/bridge/src/AbsContractAdapter.ts (hunk 3 above)
Comment (manh-vv) [reply packages/bridge/src/AbsContractAdapter.ts:32]: It is a good suggestion. I would like to have a member fix it in another issue #70

File: packages/bridge/src/ContractAdapter.ts (hunk 4 above)
Comment (manh-vv) [reply packages/bridge/src/ContractAdapter.ts:15]: The optional setting here means we don't have to have that property in a Typescript object.

**Review by manh-vv — COMMENTED** [review]
updated

**Pull request merged**